test/volume_test.go

```diff
@@ -22,38 +25,61 @@ type VolumeWorkflowSuite struct {
 	volcollService      *service.VolumeCollectionService
 	initiatorGrpService *service.InitiatorGroupService
 }
-
+func (suite *VolumeWorkflowSuite) SetupTest() {
+    fmt.Println("SetupTest")
+	if testResult == "PASS" && result == "PASS" {
+	testResult = "PASS"	
+	} else {
+		testResult = "FAIL"
+	}
+	fmt.Println(testResult)    
+}
 func (suite *VolumeWorkflowSuite) SetupSuite() {
+	fmt.Printf("setting up suite\n")
 	groupService, err := config()
-	assert.Nilf(suite.T(), err, "Unable to connect to group, err: %v", err)
+	require.Nilf(suite.T(), err, "Unable to connect to group, err: %v", err)
 	suite.groupService = groupService
 	suite.volumeService = groupService.GetVolumeService()
 	suite.snapshotService = groupService.GetSnapshotService()
 	suite.volcollService = groupService.GetVolumeCollectionService()
 	suite.initiatorGrpService = groupService.GetInitiatorGroupService()
 	_, err = createDefaultInitiatorGrp(suite.initiatorGrpService)
-	assert.Nilf(suite.T(), err, "Unable to create default initiator group, err: %v", err)
+	require.Nilf(suite.T(), err, "Unable to create default initiator group, err: %v", err)
 	_, err = createDefaultVolColl(suite.volcollService)
-	assert.Nilf(suite.T(), err, "Unable to create default volume collection, err: %v", err)
+	require.Nilf(suite.T(), err, "Unable to create default volume collection, err: %v", err)
 	suite.createVolume()
+	
 }
 
+
 func (suite *VolumeWorkflowSuite) TearDownSuite() {
+	
+	if testResult == "PASS" && result == "PASS" {
+		testResult = "PASS"	
+	} else {
+			testResult = "FAIL"
```

Comment:
This redundant code could be eliminated if the SetupTest logic is moved to an AfterTest function.

Reply:
Refactored the code